### PR TITLE
`fjson` and fuzzing cleanup

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Remove some old code. Fix `fuzztool`.

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: cmd/fuzztool/fuzztool.go [hunk 1]
```diff
@@ -227,19 +227,20 @@ func main() {
 		case "generated":
 			src = generatedCorpus
 		default:
-			src, err = filepath.Abs(src)
+			src, err = filepath.Abs(cli.Corpus.Src)
```
Comment (AlekSi) [inline cmd/fuzztool/fuzztool.go:230]: It was broken since #1159

File: internal/bson/bson_test.go [hunk 2]
```diff
@@ -182,6 +196,19 @@ func fuzzBinary(f *testing.F, testCases []testCase, newFunc func() bsontype) {
...
+
+		// Test that generated value can be marshaled for logging.
+		// Currently, that seems to be the best place to check it since generating values from BSON bytes is very easy.
+		{
+			// not a "real" type
+			if _, ok := v.(*CString); ok {
+				t.Skip()
+			}
+
+			mB, err := fjson.Marshal(fromBSON(v))
```
Comment (AlekSi) [inline internal/bson/bson_test.go:208]: @chilagrow that's what I was thinking of

Comment (chilagrow) [reply internal/bson/bson_test.go:208]: @AlekSi yeah I've been reading this PR :pray:  I see what you did with tests that used CString, your comments explains a lot to me thanks!

**Comment by chilagrow** [comment]
Thanks for the docs!
@AlekSi Fuzz test CI is looking unhappy, doesn't seem like it's related to this PR though https://github.com/FerretDB/FerretDB/actions/runs/3262667842/jobs/5360097653

**Comment by AlekSi** [comment]
Yeah, it fails due to #364

**Review by w84thesun — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by noisersup — APPROVED** [review]

**Comment by AlekSi** [comment]
Ok, "Fuzz short" is not required for PRs to be merged until we re-add validation and fix all fuzzing issues.

**Pull request merged**